smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -378,6 +380,11 @@ boolean hasTooManyMessagesWithoutAck() {
             return false;
         }
 
+        private boolean isStillAssigned() {
+            // If the topic/partition is not assigned to us already, the store will be cleared eventually.
+            return assignments.contains(topicPartition);
```

Comment:
Don't `offsetStores` include the same information? In other words, could this method be implemented as `return offsetStore.containsKey(topicPartition);`?

Reply:
No, because the deletion happens asynchronously, and so there is a window where it would still be there, while not being assigned anymore.

Reply:
Ah interesting!